Roys/app/src/main/java/com/dwg_karrier/roys/Authentication.java

```diff
@@ -0,0 +1,248 @@
+      } catch (JSONException e) {
+        Log.e("JSON Parser", "Error parsing data " + e.toString());
+      }
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return jObj;
+  }
+
+  private class TokenGet extends AsyncTask<String, String, JSONObject> {
+    String code;
+    private ProgressDialog pDialog;
+    Context mainContext;
+
+    public TokenGet(String tokencode, Context context) {
+      code = tokencode;
+      mainContext = context;
+    }
+
+    @Override
+    protected void onPreExecute() {
+      super.onPreExecute();
+      pDialog = new ProgressDialog(mainContext);
+      pDialog.setMessage("Contacting Feedly ...");
+      pDialog.setIndeterminate(false);
+      pDialog.setCancelable(true);
+      pDialog.show();
+    }
+
+    @Override
+    protected JSONObject doInBackground(String... args) {
+      JSONObject json = requestToken(TOKENURL, code, CLIENTID, CLIENTSECRET, REDIRECTURI, GRANTTYPE);
+      return json;
+    }
+
+    @Override
+    protected void onPostExecute(JSONObject json) {
+      if (json != null) {
+        try {
+          final String URL = "https://cloud.feedly.com/v3/streams/contents?streamId=user/" + json.getString("id") + "/category/global.all";
+          new GetPageList(new DataBaseOpenHelper(mainContext), mainContext, json.getString("access_token"),pDialog).execute(URL);
+        } catch (JSONException e) {
+          e.printStackTrace();
+        }
+      } else {
+        Toast.makeText(mainContext, "Network Error", Toast.LENGTH_SHORT).show();
+        pDialog.dismiss();
+      }
+    }
+  }
+
+  private class GetPageList extends AsyncTask<String, Void, String> {
+    private DataBaseOpenHelper dataBaseOpenHelper;
+    private Context mainContext;
+    String accessToken;
+    private ProgressDialog pDialog;
+
+    public GetPageList(DataBaseOpenHelper dbHelper, Context context, String token, ProgressDialog dialog) {
+      dataBaseOpenHelper = dbHelper;
+      mainContext = context;
+      accessToken = token;
+      pDialog = dialog;
+    }
+
+    private int countWords(String html) throws Exception {
+      org.jsoup.nodes.Document dom = Jsoup.parse(html);
+      String text = dom.text();
+
+      return text.split(" ").length;
+    }
+
+    @Override
+    protected String doInBackground(String... params) {
+      try {
+        pDialog.setMessage("Bring Scripted Pages from Feedly ...");
+        URL url = new URL(params[0]);
+        HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+        urlConnection.setRequestProperty("Authorization", "OAuth " + accessToken);
+
+        InputStream stream = new BufferedInputStream(urlConnection.getInputStream());
+        String result = convertStreamToString(stream);
+
+        JSONObject obj = new JSONObject(result);
+        JSONArray arr = obj.getJSONArray("items");
+        int len = arr.length();
```

Comment:
I think the `len` is a variable. because the `arr` is array of scripted pages. In that case, should we use declare the len with final?